astropy/cosmology/parameter/tests/test_parameter.py

```diff
@@ -486,3 +455,23 @@ def test_Parameter_repr_roundtrip(self, param):
         NP = eval(repr(P))  # Evaluate string representation back into a param.
 
         assert P == NP
+
+    # ========================================================================
+
+    def test_make_from_Parameter(self, cosmo_cls, clean_registry):
+        """Test the parameter creation process. Uses ``__set__``."""
+
+        class Example(cosmo_cls):
```

Comment:
Is this class needed? Isn't it possible to use instances of `cosmo_cls`?

Reply:
`cosmo_cls` can be abstract, in particular the property `is_flat`. So this is unfortunately necessary.